src/Controls/tests/TestCases.HostApp/Issues/Issue28986.xaml

```diff
@@ -0,0 +1,128 @@
+                           FontSize="14"
+                           HorizontalOptions="Center"
+                           TextColor="White" />
+                    
+                    <Label x:Name="CurrentSettingsLabel"
+                           Text="Current MainGrid SafeAreaEdges: All"
+                           FontSize="16"
+                           FontAttributes="Bold"
+                           HorizontalOptions="Center"
+                           TextColor="White"
+                           AutomationId="CurrentSettings" />
+                    
+                    <!-- Simple button controls for MainGrid -->
+                    <Border BackgroundColor="White" Stroke="Gray" StrokeThickness="2" Padding="20" Margin="10">
+                        <VerticalStackLayout Spacing="15">
+                            <Label Text="MainGrid Safe Area Control:" 
+                                   FontSize="18" 
+                                   FontAttributes="Bold"
+                                   HorizontalOptions="Center" />
+                            
+                            <Label Text="Test different SafeAreaEdges values on the MainGrid"
+                                   FontSize="12"
+                                   HorizontalOptions="Center"
+                                   TextColor="Gray" />
+                            
+                            <Button Text="Set MainGrid to None (Edge-to-edge)"
+                                    Clicked="OnGridSetNoneClicked"
+                                    BackgroundColor="LightBlue"
+                                    AutomationId="GridResetNoneButton" />
+                            
+                            <Button Text="Set MainGrid to Container (Respect notches/bars)"
+                                    Clicked="OnGridSetContainerClicked"
+                                    BackgroundColor="LightGreen"
+                                    AutomationId="GridSetContainerButton" />
+                            
+                            <Button Text="Set MainGrid to All (Full safe area)"
+                                    Clicked="OnGridSetAllClicked"
+                                    BackgroundColor="LightCoral"
+                                    AutomationId="GridResetAllButton" />
+                            
+                            <!-- Per-edge controls for specific scenarios -->
+                            <Label Text="Per-Edge Controls:" 
+                                   FontSize="16" 
+                                   FontAttributes="Bold"
+                                   HorizontalOptions="Center"
+                                   Margin="0,10,0,0" />
+                            
+                            <Button Text="Set Bottom Edge to SoftInput"
```

Comment:
You're going to need to add an entry control that you can focus to validate that "SoftInput" is working for the tests